tools/test/h5repack/CMakeLists.txt

```diff
@@ -27,19 +27,19 @@ set (REPACK_COMMON_SOURCES
     ${HDF5_TOOLS_SRC_H5REPACK_SOURCE_DIR}/h5repack_verify.c
     ${HDF5_TOOLS_SRC_H5REPACK_SOURCE_DIR}/h5repack.c
 )
+# Add h5repack test executable
 add_executable (h5repacktest ${REPACK_COMMON_SOURCES} ${HDF5_TOOLS_TEST_H5REPACK_SOURCE_DIR}/h5repacktst.c)
 target_include_directories (h5repacktest
     PRIVATE "${HDF5_TOOLS_SRC_H5REPACK_SOURCE_DIR};${HDF5_TOOLS_ROOT_DIR}/lib;${HDF5_SRC_INCLUDE_DIRS};${HDF5_TEST_SRC_DIR};${HDF5_SRC_BINARY_DIR};$<$<BOOL:${HDF5_ENABLE_PARALLEL}>:${MPI_C_INCLUDE_DIRS}>"
 )
-if (BUILD_STATIC_LIBS)
```

Comment:
I think this is an issue where we need to determine the purpose of `HDF5_BUILD_STATIC_TOOLS` and align between it and `BUILD_STATIC_LIBS`. Since this is a regular test program, it should probably be using `BUILD_STATIC_LIBS` as before rather than `HDF5_BUILD_STATIC_TOOLS`. But, when it comes time to link against the tools library we need to check `HDF5_BUILD_STATIC_TOOLS` to link against the correct one (though we should probably solve this higher up and just give lower level CMake code a single target to link against unconditionally).